Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1046,8 +1046,16 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
           position: position,
           .consecutiveDeclarationsOnSameLine,
           fixIts: [
-            FixIt(message: .insertSemicolon, changes: .makePresent(semicolon))
-          ],
+            node.lastToken(viewMode: .sourceAccurate).map {
+              FixIt(
+                message: .insertNewline,
+                changes: [
+                  .replaceTrailingTrivia(token: $0, newTrivia: .newlines(1))
```

Comment:
We should be retaining the existing trivia as well because it could be a comment. If you want to trim trailing whitespace before the newline, you should make sure that you’re not removing comments.

Reply:
Fixed and added testcase